proxy/src/http/org.rs

```diff
@@ -288,81 +258,74 @@ where
-    pub async fn get<R: registry::Client>(
-        registry: http::Shared<R>,
-        org_id: String,
-    ) -> Result<impl Reply, Rejection> {
-        let reg = registry.read().await;
-        let org_id = registry::Id::try_from(org_id).map_err(Error::from)?;
-        let org = reg.get_org(org_id).await?;
+    pub async fn get<R>(ctx: http::Ctx<R>, org_id: String) -> Result<impl Reply, Rejection>
+    where
+        R: registry::Client,
+    {
+        let ctx = ctx.read().await;
+        let org_id = registry::Id::try_from(org_id).map_err(error::Error::from)?;
+        let org = ctx.registry.get_org(org_id).await?;
 
         Ok(reply::json(&org))
     }
 
     /// Register a project in the Registry.
-    pub async fn register_project<R: registry::Client>(
-        registry: http::Shared<R>,
-        subscriptions: notification::Subscriptions,
+    pub async fn register_project<R>(
+        ctx: http::Ctx<R>,
         org_id: String,
         project_name: String,
         input: http::RegisterProjectInput,
-    ) -> Result<impl Reply, Rejection> {
-        http::register_project(
-            registry,
-            subscriptions,
-            registry::DomainType::Org,
-            org_id,
-            project_name,
-            input,
-        )
-        .await
+    ) -> Result<impl Reply, Rejection>
+    where
+        R: registry::Client,
+    {
+        http::register_project(ctx, registry::DomainType::Org, org_id, project_name, input).await
     }
 
     /// Get the [`registry::Project`] under the given org id.
-    pub async fn get_project<R: registry::Client>(
-        registry: http::Shared<R>,
+    pub async fn get_project<R>(
+        ctx: http::Ctx<R>,
         org_id: String,
         project_name: String,
-    ) -> Result<impl Reply, Rejection> {
-        let reg = registry.read().await;
-        let org_id = registry::Id::try_from(org_id).map_err(Error::from)?;
+    ) -> Result<impl Reply, Rejection>
+    where
+        R: registry::Client,
+    {
+        let ctx = ctx.read().await;
+        let org_id = registry::Id::try_from(org_id).map_err(error::Error::from)?;
         let project_domain = registry::ProjectDomain::Org(org_id);
-        let project_name = registry::ProjectName::try_from(project_name).map_err(Error::from)?;
-        let project = reg.get_project(project_domain, project_name).await?;
+        let project_name =
+            registry::ProjectName::try_from(project_name).map_err(error::Error::from)?;
+        let project = ctx
+            .registry
+            .get_project(project_domain, project_name)
+            .await?;
 
         Ok(reply::json(&project))
     }
 
     /// Get all projects under the given org id.
-    pub async fn get_projects<R>(
-        registry: http::Shared<R>,
-        peer: Arc<Mutex<coco::PeerApi>>,
-        org_id: String,
-    ) -> Result<impl Reply, Rejection>
+    pub async fn get_projects<R>(ctx: http::Ctx<R>, org_id: String) -> Result<impl Reply, Rejection>
```

Comment:
I wonder if we should turn all these `String`s into types while we're doing this refactor?

Reply:
Would do that in a follow-up, there are more improvements to come.